docs/source/ttnn/dependencies/tt_lib.rst

```diff
@@ -58,40 +63,47 @@ Or to add one with a member:
         }
     };
 
-And below, is an example of how to declare a new on-host operation with all of the methods required by the interface.
+
+New Device Operation with optional output tensors
+-------------------------------------------------
+
+If an operation is expected to leverage optional output tensors, please use instead the validate_with_output_tensors
+and create_output_tensors with the additional parameter for the output_tensors.
 
 .. code-block::
 
     struct <NewOperation> {
-        void validate(const std::vector<Tensor> &input_tensors) const;
+        void validate_with_output_tensors(const std::vector<Tensor> &input_tensors, const std::vector<std::optional<Tensor>>& output_tensors) const;
```

Comment:
Oh man...

Reply:
No, we still have the current validate. I just re-arranged the order to have `HostOperation` at the end